lib/workspace.rb

```diff
@@ -0,0 +1,45 @@
+module SlackBot
+  class Workspace
+    attr_reader :users, :channels, :selected
+
+    def initialize
+      @users = SlackBot::User.list
+      @channels = SlackBot::Channel.list
+      @selected = nil
+    end
+
+    def list_users
+      user_list = @users.map do |user|
+        user.details
+      end
+      return user_list
+    end
+
+    def list_channels
+      channel_list = @channels.map do |channel|
+        channel.details
+      end
+      return channel_list
+    end
+
+    def select_user(input)
+      found_user = @users.find do |user|
+        input == user.id || input == user.name
+      end
+      @selected = found_user
```

Comment:
If you aren't going to do additional validation you can just directly assign `@selected` to `found_user`.